File: Quantum Anealing/src/quantum/anealing/TestQuantumAnnealingAlgorithm.java
package quantum.anealing;

import main.model.Vertex;
import main.model.Graph;
import java.util.List;

public class TestQuantumAnnealingAlgorithm {

    private static final float COST_REDUCTION_FACTOR = 0.75f;
    private static final int TROTTER_REPLICAS = 50;     // P
    private static final float TEMPERATURE = 100f;         // T
    private static final int MONTE_CARLO_STEP = 100;   // M
    private static final float TUNNLING_FIELD_INITIAL = 1f;
    private static final float TUNNLING_FIELD_FINAL = .5f;
    private static final float TUNNLING_FIELD_EVAPORATION = .9f;

    public void execute(
            Graph graph,
            List<Vertex> candidateSinks,
            List<Vertex> candidateControllers,
            int sensorSinkMaxDistance,
            int sensorControllerMaxDistance,
            int maxSinkCoverage,
            int maxControllerCoverage,
            int maxSinkLoad,
            int maxControllerLoad,
            int costSink,
            int costController) {

        QuantumAnealing qa = new QuantumAnealing(
                graph,
                candidateSinks,
                candidateControllers,
                sensorSinkMaxDistance,
                sensorControllerMaxDistance,
                maxSinkCoverage,
                maxControllerCoverage,
                maxSinkLoad,
                maxControllerLoad,
                costSink,
                costController,
                COST_REDUCTION_FACTOR,
                TROTTER_REPLICAS,
                TEMPERATURE,
                MONTE_CARLO_STEP,
                TUNNLING_FIELD_INITIAL,
                TUNNLING_FIELD_FINAL,
                TUNNLING_FIELD_EVAPORATION
        );

        qa.execute();
    }
}

File: Quantum Anealing/Python/quantum_annealing/test_quantum_annealing_algorithm.py
from .quantum_annealing import QuantumAnnealing


class TestQuantumAnnealingAlgorithm:
    def __init__(self,
                 graph,
                 candidate_sinks,
                 candidate_controllers,
                 sink_y_spin_variables,
                 controller_y_spin_variables,
                 sensor_sink_max_distance,
                 sensor_controller_max_distance,
                 max_sink_coverage,
                 max_controller_coverage,
                 max_sink_load,
                 max_controller_load,
                 cost_sink,
                 cost_controller
                 ):
        self.COST_REDUCTION_FACTOR = 0.75
        self.TROTTER_REPLICAS = 50
        self.TEMPERATURE = 100.0
        self.MONTE_CARLO_STEP = 100
        self.TUNNLING_FIELD_INITIAL = 1.0
        self.TUNNLING_FIELD_FINAL = .5
        self.TUNNLING_FIELD_EVAPORATION = .95

        self.qa = QuantumAnnealing(graph,
                                   candidate_sinks,
                                   candidate_controllers,
                                   sink_y_spin_variables,
                                   controller_y_spin_variables,
                                   sensor_sink_max_distance,
                                   sensor_controller_max_distance,
                                   max_sink_coverage,
                                   max_controller_coverage,
                                   max_sink_load,
                                   max_controller_load,
                                   cost_sink,
                                   cost_controller,
                                   self.COST_REDUCTION_FACTOR,
                                   self.TROTTER_REPLICAS,
                                   self.TEMPERATURE,
                                   self.MONTE_CARLO_STEP,
                                   self.TUNNLING_FIELD_INITIAL,
                                   self.TUNNLING_FIELD_FINAL,
                                   self.TUNNLING_FIELD_EVAPORATION)

    def execute(self):
        return self.qa.execute()
